ResearchKit/Charts/ORKGraphChartView.m

```diff
@@ -693,14 +748,20 @@ - (void)updateScrubberLineAccessories:(CGFloat)xPosition {
                                       CGRectGetMinY(_scrubberLine.frame),
                                       textSize.width + ScrubberLabelHorizontalPadding,
                                       textSize.height + ScrubberLabelVerticalPadding);
+
+    if (scrubbingValue == ORKCGFloatInvalidValue) {
+        [self setScrubberAccessoryViewsHidden:YES];
+    } else {
+        [self setScrubberAccessoryViewsHidden:NO];
+    }
 }
 
-- (CGFloat)snappedXPosition:(CGFloat)xPosition {
+- (CGFloat)snappedXPosition:(CGFloat)xPosition plotIndex:(NSInteger)plotIndex {
     CGFloat numberOfXAxisPoints = self.numberOfXAxisPoints;
     CGFloat widthBetweenPoints = CGRectGetWidth(_plotView.frame) / numberOfXAxisPoints;
-    for (NSUInteger positionIndex = 0; positionIndex < ((NSArray *)_dataPoints[0]).count; positionIndex++) {
+    for (NSUInteger positionIndex = 0; positionIndex < ((NSArray *)_dataPoints[plotIndex]).count; positionIndex++) {
```

Comment:
Declare `NSInteger positionCount = ((NSArray *)_dataPoints[plotIndex]).count;` and use it in the for loop?